android/src/main/java/com/reactnativeteadssdkmodule/RNInReadAdPlacement.kt

```diff
@@ -0,0 +1,198 @@
+          }
+        }
+
+        override fun onAdImpression() {
+          val params = Arguments.createMap()
+          params.putString("didRecordImpression", "didRecordImpression")
+          sendEvent(reactApplicationContext, "didRecordImpression", params)
+          Handler(Looper.getMainLooper()).post {
+            Log.d(
+              "from ad",
+              "didRecordImpression"
+            )
+          }
+        }
+
+        override fun onAdExpandedToFullscreen() {
+          val params = Arguments.createMap()
+          params.putString("didExpandedToFullscreen", "didExpandedToFullscreen")
+          sendEvent(reactApplicationContext, "didExpandedToFullscreen", params)
+          Handler(Looper.getMainLooper()).post {
+            Log.d(
+              "from ad",
+              "didExpandedToFullscreen"
+            )
+          }
+        }
+
+        override fun onAdCollapsedFromFullscreen() {
+          val params = Arguments.createMap()
+          params.putString("didCollapsedFromFullscreen", "didCollapsedFromFullscreen")
+          sendEvent(reactApplicationContext, "didCollapsedFromFullscreen", params)
+          Handler(Looper.getMainLooper()).post {
+            Log.d("from ad", "didCollapsedFromFullscreen")
+          }
+        }
+
+        override fun onAdRatioUpdate(adRatio: AdRatio) {
+          val params = Arguments.createMap()
+          params.putString("didUpdateRatio", "didUpdateRatio")
+          sendEvent(reactApplicationContext, "didUpdateRatio", params)
+          Handler(Looper.getMainLooper()).post {
+            Log.d(
+              "from ad",
+              "didUpdateRatio"
+            )
+
+          }
+        }
+
+        override fun onFailToReceiveAd(failReason: String) {
+          val params = Arguments.createMap()
+          params.putString("didFailToReceiveAd", "didFailToReceiveAd")
+          sendEvent(reactApplicationContext, "didFailToReceiveAd", params)
+          Handler(Looper.getMainLooper()).post {
+            Log.d(
+              "from Ad",
+              "didFailToReceiveAd"
+            )
+            Log.d("from fail", failReason)
+          }
+        }
+      },
+      object : VideoPlaybackListener {
+        override fun onVideoComplete() {
+          val params = Arguments.createMap()
+          params.putString("didComplete", "didComplete")
+          sendEvent(reactApplicationContext, "didComplete", params)
+          Handler(Looper.getMainLooper()).post {
+            Log.d(
+              "from ad",
+              "didComplete"
+            )
+          }
+        }
+
+        override fun onVideoPause() {
+          val params = Arguments.createMap()
+          params.putString("didPause", "didPause")
+          sendEvent(reactApplicationContext, "didPause", params)
+          Handler(Looper.getMainLooper()).post {
+            Log.d(
+              "from ad",
+              "didPause"
+            )
+          }
+        }
+
+        override fun onVideoPlay() {
```

Comment:
I see that in general you are using iOS signature names onVideoPlay —> didPlay.

In the end from the publisher perspective distributing from RN to Android or iOS, do they have to listen the callbacks from Android and iOs separately or does they subscribe to a single listener?